src/reducers/port.reducer.js

```diff
@@ -118,21 +117,72 @@ function setPort(state: State, port: Port) {
 	return state;
 }
 
-export default function portReducer(state: State, action: PortAction): State {
+function addPortDeprecated(action, state) {
+	if (!state.getIn(['nodes', action.nodeId])) {
+		invariant(false, `Can't set a new port ${action.id} on non existing node ${action.nodeId}`);
+	}
+	return setPort(state, {
```

Comment:
i think spread is not explicit about what is passed to the function parameter, it would also provide unwanted action metadata and type wich would add noise in case of debugging or error report.